src/builder/Errors.sol

```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.23;
+
+/// Library of shared errors used across Quark Builder files
+library Errors {
```

Comment:
nit: Curious, should we also move other builder errors to here. (probably in future refactor PRs)